src/component/Timer/index.jsx

```diff
@@ -45,6 +44,25 @@ const Timer = () => {
     clearTimeout(tickTimeoutId.current);
   };
 
+  const handleResetClick = () => {
+    setIsRunning(false);
+    clearTimeout(tickTimeoutId.current);
+    setTime([currentPeriod.mins, currentPeriod.secs]);
+  };
+
+  const handleNextClick = () => {
+    const nextPeriod = determineNextPeriod(currentPeriod, counter);
+    clearTimeout(tickTimeoutId.current);
+    setTime([nextPeriod.mins, nextPeriod.secs]);
+    setCurrentPeriod(nextPeriod);
+    setIsRunning(false);
+    setCounter(counter + 1);
+    timeOverSoundAudio.play();
+    if (counter === 6) {
+      setCounter(-1);
+    }
```

Comment:
This looks like it's the same code used by the `reset` function. Can you share the code between those two? To learn more about where I am coming from, check out the "DRY programming principle"

Reply:
I searched about this concept, it is mean Don't repeat yourself aims to "reducing repetition of software" and this is very important concept to any one who want be professional programmer. I'm deebly gratful for you, I'm learning so many thing from you.